docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,161 @@
+---
+lang: en
+title: 'Relations'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/HasMany-relation.html
+summary:
+---
+
+## Defining a HasMany Relation
+
+Model relations in LoopBack 4 are defined by having a decorator on a designated
+relational property. This section describes how to define a `hasMany` relation
+at the model level using the `hasMany` decorator. The relation constrains the
+target repository by the foreign key property on its associated model.
+
+The definition of the `hasMany` relation is inferred by using the `hasMany`
+decorator. The decorator takes in the target model class constructor and
+optionally a custom foreign key to store the relation metadata. The decorator
+logic also designates the relation type and tries to infer the foreign key on
+the target model (or `keyTo` in the relation metadata) default value (source
+model name appended with `id` in camel case, also used in LoopBack 3). It also
+calls `property.array()` to ensure that the type of the property is inferred
+properly as an array of the target model instances.
+
+The decorated property name is used as the relation name and stored as part of
+the source model definition's relation metadata. The following example shows how
+to define a `hasMany` relation on a source model `Customer`.
+
+{% include code-caption.html content="/src/models/customer.model.ts" %}
+
+```ts
+import {Order} from './order.model.ts';
+import {Entity, property, hasmany} from '@loopback/repository';
+
+export class Customer extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+  })
+  name: string;
+
+  @hasMany(Order) orders: Order[];
+}
+```
+
+The metadata generated for the above decoration is as follows:
+
+```js
+{
+  type: 2; // denotes hasMany relation (this is an enum value)
+  keyTo: customerId; // Order model should have this property declared
+}
+```
+
+The property type metadata is also preserved as an array of type `Order` as part
+of the decoration. Another usage of the decorator with a custom foreign key name
+for the above example is as follows:
+
+```ts
+// inside Customer model class
+@hasMany(Order, {keyTo: 'custId'}) orders: Order[];
+```
+
+## Configuring a HasMany relation
+
+The configuration and resolution of a `HasMany` relation takes place at the
+repository level. Once `HasMany` relation is defined on the source model, then
+there are a couple of steps involved to configure it and use it. On the source
+repository, the following are required:
+
+- Use [Dependency Injection](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<TargetModel, typeof TargetModel.prototype.id>` on
```

Comment:
Catch!

The factory is defined as follows (see [relation.factory.ts](https://github.com/strongloop/loopback-next/blob/fe681cad645708c30318d6586da7b3b814b4fbfd/packages/repository/src/repositories/relation.factory.ts#L14)) - the second argument is ID type of the **source** model.

```ts
export type HasManyRepositoryFactory<Target extends Entity, SourceID> = (
  fkValue: SourceID,
) => HasManyRepository<Target>;
```

We should fix this in our codebase too, see e.g. acceptance tests like [has-many.relation.acceptance.ts](https://github.com/strongloop/loopback-next/blob/fe681cad645708c30318d6586da7b3b814b4fbfd/packages/repository/test/acceptance/has-many.relation.acceptance.ts#L207).

While we are making those changes, I think it would be great to change the order of the template arguments, so that we start with source-related types and follow with target-related types. It feels more natural, at least to me. Thoughts?